core/src/components/tooltip/tooltip.tsx

```diff
@@ -26,112 +26,57 @@ export class TdsTooltip {
-
-  componentDidLoad() {
-    this.target = this.referenceEl ?? document.querySelector(this.selector);
-
-    const thisValue = this;
-    this.popperInstance = createPopper(this.target, this.tooltip, {
-      placement: thisValue.placement,
-      modifiers: [
-        {
-          name: 'positionCalc',
-          enabled: true,
-          phase: 'main',
-          fn({ state }) {
-            if (state.placement === 'bottom-start' || state.placement === 'right-start') {
-              thisValue.border = 'top-left';
-            } else if (state.placement === 'bottom-end' || state.placement === 'left-start') {
-              thisValue.border = 'top-right';
-            } else if (state.placement === 'top-end' || state.placement === 'left-end') {
-              thisValue.border = 'bottom-right';
-            } else if (state.placement === 'top-start' || state.placement === 'right-end') {
-              thisValue.border = 'bottom-left';
-            } else if (state.placement === 'bottom' || state.placement === 'top') {
-              thisValue.border = 'default';
-            }
-          },
-        },
-        {
-          name: 'offset',
-          options: {
-            offset: [this.offsetSkidding, this.offsetDistance],
-          },
-        },
-      ],
-    });
-
-    const showTooltip = () => {
-      this.show = true;
-    };
-
-    const hideTooltip = () => {
-      this.show = false;
-    };
-
-    // For tabbing over element
-    this.target.addEventListener('focusin', () => {
-      showTooltip();
-    });
-
-    this.target.addEventListener('focusout', () => {
-      hideTooltip();
-    });
-
-    // For hovering over element with selector
-    this.target.addEventListener('mouseenter', () => {
-      showTooltip();
-    });
-
-    this.target.addEventListener('mouseleave', () => {
-      hideTooltip();
-    });
-
-    // For hovering over Tooltip itself:
-    if (this.mouseOverTooltip === true) {
-      this.tooltip.addEventListener('mouseenter', () => {
-        showTooltip();
-      });
-
-      this.tooltip.addEventListener('mouseleave', () => {
-        hideTooltip();
-      });
-    }
-  }
-
-  disconnectedCallback() {
-    this.popperInstance?.destroy();
-  }
+  popperjsExtraModifiers = [
+    {
+      name: 'positionCalc',
+      enabled: true,
+      phase: 'main',
+      fn: ({ state }) => {
+        if (state.placement === 'bottom-start' || state.placement === 'right-start') {
+          this.border = 'top-left';
+        } else if (state.placement === 'bottom-end' || state.placement === 'left-start') {
+          this.border = 'top-right';
+        } else if (state.placement === 'top-end' || state.placement === 'left-end') {
+          this.border = 'bottom-right';
+        } else if (state.placement === 'top-start' || state.placement === 'right-end') {
+          this.border = 'bottom-left';
+        } else if (state.placement === 'bottom' || state.placement === 'top') {
+          this.border = 'default';
```

Comment:
Could this maybe be a switch case? Still learning when to use and not use them so might be wrong :)

Reply:
Doesn't need to be part of this PR, mostly just curious ☺️

Reply:
Yes it could be!
It comes down to readability in this case.
The equivalent switch statement is more lines but maybe more readable?

```js
switch(state.placement) {
  case 'bottom-start':
  case 'right-start':
    this.border = 'top-left';
    break;
  case 'bottom-end':
  case 'left-start':
    this.border = 'top-right';
    break;
  case 'top-end':
  case 'left-end':
    this.border = 'bottom-right';
    break;
  case 'top-start':
  case 'right-end':
    this.border = 'bottom-left';
    break;
  case 'bottom':
  case 'top':
  default:
    this.border = 'default';
}
```

Reply:
```js
[
  ['top-left', ['bottom-start', 'right-start']],
  ['top-right', ['bottom-end', 'left-start']],
  ['bottom-right', ['top-end', 'left-end']],
  ['bottom-left', ['top-start', 'right-end']],
  ['default', ['bottom', 'top']]
].forEach(([border, placements]) => {
  if(placements.includes(state.placement)) {
    this.border = border;
  } else {
    this.border = 'default';
  }
})
```
Another alternative perhaps